### PR TITLE
release(v0.12.3): prepare release

### PR DESCRIPTION
**Pull request by AlekSi**
This is the official v0.12.3 release.

---
This change is [Reviewable](https://reviewable.io/reviews/talos-systems/talos/4297)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,398 @@
...
+
+`talosctl upgrade-k8s` now checks if cluster has any resources which are going to be removed or migrated to the new version after upgrade
+and shows that as a warning before the upgrade.
+Additionally, `upgrade-k8s` command now has `--dry-run` flag that only prints out warnings and upgrade summary.
+
+
+### Contributors
+
+* Andrey Smirnov
+* Andrey Smirnov
```
Comment (AlekSi) [inline CHANGELOG.md:104]: We probably should do some deduplication/normalisation

File: go.mod [hunk 2]
```diff
@@ -80,7 +80,7 @@ require (
 	github.com/talos-systems/go-smbios v0.1.0
 	github.com/talos-systems/grpc-proxy v0.2.0
 	github.com/talos-systems/net v0.3.0
-	github.com/talos-systems/talos/pkg/machinery v0.0.0-00010101000000-000000000000
+	github.com/talos-systems/talos/pkg/machinery v0.12.3
```
Comment (smira) [inline go.mod:83]: :+1:

File: hack/protoc-lint/go.mod [hunk 3]
```diff
@@ -7,6 +7,6 @@ replace github.com/talos-systems/talos/pkg/machinery => ../../pkg/machinery
 require (
 	github.com/stretchr/testify v1.7.0
 	github.com/talos-systems/talos/pkg/machinery v0.0.0-00010101000000-000000000000
-	google.golang.org/grpc v1.39.0
+	google.golang.org/grpc v1.40.0
```
Comment (smira) [inline hack/protoc-lint/go.mod:10]: was this intentional?

**Review by smira — APPROVED** [review]

**Comment by AlekSi** [comment]
https://ci.dev.talos-systems.io/talos-systems/talos/20474
https://ci.dev.talos-systems.io/talos-systems/talos/20475

**Pull request merged**